content/en/docs/contributing/third-party-code-donation.md

```diff
@@ -0,0 +1,80 @@
+and containerd.
+
+## Requirements
+
+- Code must be licensed appropriately, including any dependencies   
+  We'd prefer [Apache 2.0](https://tldrlegal.com/license/apache-license-2.0-(apache-2.0)) since that's
+  what cert-manager [uses](https://github.com/jetstack/cert-manager/blob/master/LICENSE), but the
+  license must be [OSI approved](https://opensource.org/licenses).
+- Code must conform to CNCF standards and due diligence requirements   
+  You don't need to go over this with a fine-toothed comb; the intent here is that no code donation
+  should have a negative effect on cert-manager's progress as a CNCF project. See the
+  [CNCF due diligence template](https://github.com/cncf/toc/blob/main/process/dd-review-template.md)
+- Must have a sponsor by an existing maintainer   
+  An existing regular contributor to cert-manager must sponsor the adoption of any third party code
+  donation. This ensures that there's a single point of contact for the party donating the code.
+- Must pass cert-manager conformance tests   
+  This might not apply to all donations, but where conformance tests exist any donated code must
+  pass them. E.g. for [external issuers](https://github.com/jetstack/cert-manager/blob/dffbf391dbb0fc6c1cfea62e561a9c6f54362ab0/test/e2e/suite/conformance/certificates/external/external.go#L41-L62)
+- Must provide a point-of-contact for questions about the project for at least 3 months after acceptance
+  We don't anticipate that we'd need to reach out often after the donation has been accepted,
+  but it's important to have someone we can reach out to if we need to.
+- The donation must be a defined extension type or justify why it doesn't belong in the main repositories   
+  E.g. an ACME DNS solver, a custom issuer or an ACME HTTP solver
+- Code must have a similar level of quality to cert-manager itself
+  This could be enforced by, for example, running static analysis tools on the code base similar to
+  those used by cert-manager.
+- Code must have a non-trivial test suite, including both unit tests and end-to-end tests
+  These tests must be able to be run in their entirety after a PR is raised against the repo. We don't
+  need 100% code coverage, but there should be tests for important functionality.
+- The project must adopt the cert-manager security policy and link back to the policy, as in e.g.
+  [the istio-csr `SECURITY.md`](https://github.com/cert-manager/istio-csr/blob/master/SECURITY.md)
+- Must have DCO sign-offs for all commits in place to ensure the code can legally be donated,
+  or else must have a positive DCO affirmation made by each contributor prior to donation. See below.
+
+## Preferences
+
+These items are not absolutely necessary but they definitely help if a code donation is to be accepted.
+
+- Must be written in Go   
+  We don't _need_ code to be written in Go, but we'd much prefer that it is. Since cert-manager itself
```

Comment:
```suggestion
- Should be written in Go   
  We don't _need_ code to be written in Go, but we'd much prefer that it is. Since cert-manager itself
```